src/AutoRest.CSharp/Common/Generation/Writers/RestClientWriter.cs

```diff
@@ -17,6 +19,9 @@ namespace AutoRest.CSharp.Generation.Writers
 {
     internal class RestClientWriter
     {
+        internal delegate void WriteFuncBody(CodeWriter writer, CodeWriterDeclaration messageVariable, RestClientMethod operation, string pipelineName, bool async);
+        internal delegate void WriteStatusCodeImplementation(CodeWriter writer, CodeWriterDeclaration messageVariable, RestClientMethod operation, bool async, FieldDeclaration fieldDeclaration);
```

Comment:
Well, this is not what I meant - we could change this messageVariable to `FormattableString` which is far more flexible.
```suggestion
        internal delegate void WriteFuncBody(CodeWriter writer, Formattable messageVariable, RestClientMethod operation, string pipelineName, bool async);
        internal delegate void WriteStatusCodeImplementation(CodeWriter writer, Formattable messageVariable, RestClientMethod operation, bool async, FieldDeclaration fieldDeclaration);
```

Reply:
Also this `fieldDeclaration` is quite confusing, which field is this?
Based on what I read below, this should always be the clientDiagnostics, is this true? and could you rename this parameter to make it clear?

Reply:
I know you mean `FormattableString`, but I prefer `CodeWriterDeclaration`. Any concern?

Reply:
OK, here are my thoughts:
1. if this declaration is designed to be declared outside this method (write it with `:D`), we should pass in a `FormattableString` for flexibility. Because in some circumstances, we might need to pass in an expression (like `result.Value`). This is a delegate, and it should be general
2. if this declaration is designed to be declared inside this method, we might never need to have this parameter on the delegate. The method could define its own local variable (and the code writer will ensure its name never gets collision with others)
And if this variable is also needed by other parts that followed by invocation of this method, I believe it is not very great choice to pass things around.
We will need to make some refactor after this change

Reply:
Fair enough.